reviews-loader/src/consumer.ts

```diff
@@ -0,0 +1,114 @@
+import OpenAI from 'openai'
+import { IRawReview, IReview, IReviewAnalaysis } from 'shared-types'
+import createLogger from './utils/logger'
+import reviewModel from './models/review.model'
+import winston from 'winston'
+
+import { Consumer, Kafka, EachMessagePayload } from 'kafkajs'
+
+export class ReviewsConsumer {
+  private kafkaConsumer: Consumer
+  private logger: winston.Logger
+  private openai: OpenAI
+
+  public constructor() {
+    this.kafkaConsumer = this.createKafkaConsumer()
+    this.logger = createLogger('consumer')
+    this.openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY })
+  }
+
+  public async startConsumer(): Promise<void> {
+    await this.kafkaConsumer.connect()
+    await this.kafkaConsumer.subscribe({ topic: 'reviews', fromBeginning: false })
+
+    await this.kafkaConsumer.run({
+      autoCommit: false,
+      eachMessage: async (messagePayload: EachMessagePayload) => {
+        const { topic, partition, message } = messagePayload
+        const prefix = `${topic}[${partition} | ${message.offset}] / ${message.timestamp}`
+        this.logger.info(`- ${prefix} ${message.key}#${message.value}`)
+
+        const review: IRawReview = JSON.parse(message.value!.toString())
+        this.logger.info(`Sending review to Openai... Review: ${review.value}`)
+        const response = await this.openai.chat.completions.create({
+          model: 'gpt-3.5-turbo',
+          messages: [
+            {
+              role: 'system',
+              content:
```

Comment:
Maybe put it in a static file and load it? so that we would be able to change the message without reloading the ReviewsLoader service?